x-pack/plugin/esql/tools/src/main/java/org/elasticsearch/xpack/esql/tools/ProfileParser.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.tools;
+
+import org.apache.logging.log4j.core.config.plugins.util.PluginManager;
+import org.elasticsearch.common.logging.LogConfigurator;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.json.JsonXContent;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ProfileParser {
+
+    public static void main(String[] args) throws IOException {
+        PluginManager.addPackage(LogConfigurator.class.getPackage().getName());
+        LogConfigurator.configureESLogging();
+        Logger logger = LogManager.getLogger(ProfileParser.class);
+
+        if (args.length != 2) {
+            throw new IllegalArgumentException("Requires input and output file names");
+        }
+        // Enable using the tilde shorthand `~` for the home directory.
+        Path inputFileName = Path.of(args[0].replaceFirst("^~", System.getProperty("user.home"))).toAbsolutePath();
+        Path outputFileName = Path.of(args[1].replaceFirst("^~", System.getProperty("user.home"))).toAbsolutePath();
+
+        Map<String, Object> map;
+        try (InputStream input = Files.newInputStream(inputFileName)) {
+            logger.info("Starting to parse {}", inputFileName);
+            map = XContentHelper.convertToMap(JsonXContent.jsonXContent, input, true);
```

Comment:
I wonder if we should consider model rather than a map?
Possibly something like:
```
    public record Response(Profile profile) {}
    public record Profile(List<Driver> drivers) {}
    public record Driver(
            @JsonProperty("task_description") String taskDescription,
            @JsonProperty("cluster_name") String cluster,
            @JsonProperty("node_name") String node,
            @JsonProperty("start_millis") long startMillis,
            @JsonProperty("stop_millis") long stopMillis,
            List<Operator> operators
    ) {}
    public record Operator(String operator) {}
```

Reply:
Thank you, with Jackson it's indeed much clearer. I use this approach now + updated the tests accordingly.